cli/command/image/push.go

```diff
@@ -48,12 +52,30 @@ func NewPushCommand(dockerCli command.Cli) *cobra.Command {
 	flags.BoolVarP(&opts.all, "all-tags", "a", false, "Push all tags of an image to the repository")
 	flags.BoolVarP(&opts.quiet, "quiet", "q", false, "Suppress verbose output")
 	command.AddTrustSigningFlags(flags, &opts.untrusted, dockerCli.ContentTrustEnabled())
+	flags.StringVar(&opts.platform, "platform", os.Getenv("DOCKER_DEFAULT_PLATFORM"),
+		`Push a platform-specific manifest as a single-platform image to the registry.
+'local': Platform the cli is running on
+'remote': Platform the daemon is running on
+'os[/arch[/variant]]': Explicit platform (eg. linux/amd64)`)
+	flags.SetAnnotation("platform", "version", []string{"1.45"})
 
 	return cmd
 }
 
 // RunPush performs a push against the engine based on the specified options
 func RunPush(ctx context.Context, dockerCli command.Cli, opts pushOptions) error {
+	if opts.platform != "" {
+		if _, isTty := term.GetFdInfo(dockerCli.Err()); isTty {
+			_, _ = fmt.Fprint(dockerCli.Err(), "\x1b[1;37m\x1b[1;46m[ NOTE ]\x1b[0m\x1b[0m ")
+		} else {
+			_, _ = fmt.Fprint(dockerCli.Err(), "[ NOTE ] ")
+		}
+		_, _ = fmt.Fprintln(dockerCli.Err(), `Selecting a single platform for the push operation will push the image manifest for that platform only.
+This won't push the image index/manifest list which means that other components like Buildkit attestations won't be pushed.
```

Comment:
Thinking about the wording here, we are talking about `image manifests`, `index`, `manifest list`, do we really need to be this specific? Shouldn't we have a more user-friendly terms like "single-platform image" and "multi-platform image"? 

No one really wants to know the difference between an index and a manifest list :)